org.mwc.cmap.legacy/src/MWC/TacticalData/NarrativeEntry.java

```diff
@@ -317,16 +334,26 @@ public final String toString()
 //	}
 
 
+	protected void clearHash()
+	{
+	  _hashCode = null;
+	}
+	
 	@Override
   public int hashCode()
   {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + ((_DTG == null) ? 0 : _DTG.hashCode());
-    result = prime * result + ((_entry == null) ? 0 : _entry.hashCode());
-    result = prime * result + ((_track == null) ? 0 : _track.hashCode());
-    result = prime * result + ((_type == null) ? 0 : _type.hashCode());
-    return result;
+	  if(_hashCode == null)
+	  {
+	    final int prime = 31;
+	    int result = 1;
+	    result = prime * result + ((_DTG == null) ? 0 : _DTG.hashCode());

```

Comment:
I had "random crashing" back in January, that was eventually diagnosed as incorrect hashcode implementation.  Something to do with the contract for use of a multi-threaded list class.  Aah yes, it was compareTo:
http://stackoverflow.com/a/27216587/92441

We could have more than one entry at the same DTG, so I think we really do need to use all the fields.